tools/tpm2_convert_key.c

```diff
@@ -0,0 +1,414 @@
+
+    struct key_data data;
+    convert_format format = get_format(ctx->key_output_format);
+    bool ret = false;
+
+    if (format == FMT_INVAL)
+        return false;
+    else if (! read_key(ctx->key_input_file_path, &data))
+        return false;
+
+    if (data.alg_type != TPM_ALG_RSA) {
+        LOG_ERR("Unsupported algorithm type 0x%04x in input file.",
+            data.alg_type
+        );
+
+        goto out;
+    }
+    else if (data.num_blobs != 1) {
+        LOG_ERR("Unexpected number of key blobs encountered: %zd",
+            data.num_blobs);
+
+        goto out;
+    }
+
+    ret = write_key(ctx->key_output_file_path, &data, format);
+
+out:
+    free_keys(&data);
+
+    return ret;
+}
+
+static bool init(int argc, char *argv[], tpm_convert_key_ctx *ctx) {
+
+    struct option options[] = {
+            { "keyInFile",    required_argument, NULL, 'k' },
+            { "keyOutFile",   required_argument, NULL, 'f' },
+            { "keyOutFormat", required_argument, NULL, 't' },
+            { NULL,           no_argument,       NULL, '\0' }
+    };
+
+    if (argc <=1 || argc > (int) (2 * sizeof(options) / sizeof(struct option))) {
+        showArgMismatch(argv[0]);
+        return false;
+    }
+
+    int opt;
+    while ((opt = getopt_long(argc, argv, "f:k:t:hv", options, NULL)) != -1) {
+        switch (opt) {
+        case 'f':
+            if (!optarg) {
+                LOG_ERR("Please specify the file containing the public key");
+                return false;
+            }
+            ctx->key_input_file_path = optarg;
+            break;
+        case 'k':
+            if (!optarg) {
+                LOG_ERR("Please specify the file where to save the converted public key");
+                return false;
+            }
+            ctx->key_output_file_path = optarg;
+            break;
+        case 't':
+            if (!optarg) {
+                LOG_ERR("Please specify the output file format");
+                return false;
+            }
+            ctx->key_output_format = optarg;
+            break;
+        case ':':
+            LOG_ERR("Argument %c needs a value!", optopt);
+            return false;
+        case '?':
+            LOG_ERR("Unknown Argument: %c", optopt);
+            return false;
+        default:
+            LOG_ERR("?? getopt returned character code 0%o ??", opt);
+            return false;
+        }
+    }
+
+    if (
+        !ctx->key_input_file_path ||
+        !ctx->key_output_file_path ||
+        !ctx->key_output_format) {
+        LOG_ERR("One or more required parameters missing.");
+        return false;
+    }
+
+    return true;
+}
+
+int execute_tool(int argc, char *argv[], char *envp[], common_opts_t *opts,
+        TSS2_SYS_CONTEXT *sapi_context) {
+
+    /* opts is unused, avoid compiler warning */
+    (void)opts;
+    (void)sapi_context;
+    (void)envp;
+
+    struct tpm_convert_key_ctx ctx;
+
+    memset(&ctx, 0, sizeof(struct tpm_convert_key_ctx));
```

Comment:
The downside to this is that it also writes the padding bytes (if any)... but I really don't care.